In [1]:
install.packages("ggplot2")
install.packages("datasets")
install.packages("readxl") 
install.packages("dplyr") 
install.packages("PerformanceAnalytics")
install.packages("corrplot") 
install.packages("gapminder")
install.packages("gridextra")
install.packages("Ecdat")
install.packages("corpcor")
install.packages("GPArotation")
install.packages("psych")
install.packages("IDPmisc")
install.packages("lattice") 
install.packages("treetop")
install.packages("scales")
install.packages("rcompanion")
install.packages("gmodels")
install.packages("car")
install.packages("caret")
install.packages("gvlma")
install.packages("predictmeans")
install.packages("caret")
install.packages("magrittr")
install.packages("tidyr")
install.packages("lmtest")
install.packages("popbio")
install.packages("e1071")
install.packages("data.table")
install.packages("effects")
install.packages("multcomp")
install.packages("mvnormtest")
install.packages("gapminder")
install.packages("Ecdat")

Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'ggplot2' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Warning message:
"package 'datasets' is in use and will not be installed"
Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'readxl' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)

also installing the dependency 'vctrs'




package 'vctrs' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'vctrs'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying C:\Users\sdstc\Documents\R\win-library\4.1\00LOCK\vctrs\libs\x64\vctrs.dll to C:\Users\sdstc\Documents\R\win-library\4.1\vctrs\libs\x64\vctrs.dll: Permission denied"
Warning message:
"restored 'vctrs'"


package 'dplyr' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'dplyr'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying C:\Users\sdstc\Documents\R\win-library\4.1\00LOCK\dplyr\libs\x64\dplyr.dll to C:\Users\sdstc\Documents\R\win-library\4.1\dplyr\libs\x64\dplyr.dll: Permission denied"
Warning message:
"restored 'dplyr'"



The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'PerformanceAnalytics' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'corrplot' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'gapminder' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)

Warning message:
"package 'gridextra' is not available for this version of R

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages"
Warning message:
"Perhaps you meant 'gridExtra' ?"
Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'Ecdat' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'corpcor' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'GPArotation' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'psych' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'IDPmisc' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'lattice' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)

also installing the dependencies 'sass', 'jquerylib', 'wk', 'httpuv', 'xtable', 'fontawesome', 'sourcetools', 'later', 'promises', 'bslib', 'spatstat.data', 'spatstat.utils', 'deldir', 'spatstat.random', 'spatstat.sparse', 'tensor', 'goftest', 'hexbin', 'magic', 'lpSolve', 'linprog', 'RcppProgress', 'classInt', 'rlas', 'RcppArmadillo', 'lobstr', 's2', 'units', 'lwgeom', 'shiny', 'spatstat.geom', 'spatstat.core', 'rasterVis', 'geometry', 'rgdal', 'rgeos', 'rgl', 'lidR', 'pryr', 'sf', 'stars'





  There are binary versions available but the source versions are later:
     binary source needs_compilation
rlas  1.5.3  1.6.0              TRUE
sf    1.0-7  1.0-8              TRUE

  Binaries will be installed
package 'sass' successfully unpacked and MD5 sums checked
package 'jquerylib' successfully unpacked and MD5 sums checked
package 'wk' successfully unpacked and MD5 sums checked
package 'httpuv' successfully unpacked and MD5 sums checked
package 'xtable' successfully unpacked and MD5 sums checked
package 'fontawesome' successfully unpacked and MD5 sums checked
package 'sourcetools' successfully unpacked and MD5 sums checked
package 'later' successfully unpacked and MD5 sums checked
package 'promises' successfully unpacked and MD5 sums checked
package 'bslib' successfully unpacked and MD5 sums checked
package 'spatstat.data' successfully unpacked and MD5 sums checked
package 'spatstat.utils' successfully unpacked and MD5 sums checked
package 'deldir' successfully unpacked and 

Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'scales' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'rcompanion' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'gmodels' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'car' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'caret' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'caret'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying C:\Users\sdstc\Documents\R\win-library\4.1\00LOCK\caret\libs\x64\caret.dll to C:\Users\sdstc\Documents\R\win-library\4.1\caret\libs\x64\caret.dll: Permission denied"
Warning message:
"restored 'caret'"



The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'gvlma' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'predictmeans' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'caret' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'caret'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying C:\Users\sdstc\Documents\R\win-library\4.1\00LOCK\caret\libs\x64\caret.dll to C:\Users\sdstc\Documents\R\win-library\4.1\caret\libs\x64\caret.dll: Permission denied"
Warning message:
"restored 'caret'"



The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'magrittr' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'magrittr'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying C:\Users\sdstc\Documents\R\win-library\4.1\00LOCK\magrittr\libs\x64\magrittr.dll to C:\Users\sdstc\Documents\R\win-library\4.1\magrittr\libs\x64\magrittr.dll: Permission denied"
Warning message:
"restored 'magrittr'"



The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'tidyr' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'lmtest' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'lmtest'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying C:\Users\sdstc\Documents\R\win-library\4.1\00LOCK\lmtest\libs\x64\lmtest.dll to C:\Users\sdstc\Documents\R\win-library\4.1\lmtest\libs\x64\lmtest.dll: Permission denied"
Warning message:
"restored 'lmtest'"



The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'popbio' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'e1071' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'data.table' successfully unpacked and MD5 sums checked


Warning message:
"cannot remove prior installation of package 'data.table'"
Warning message in file.copy(savedcopy, lib, recursive = TRUE):
"problem copying C:\Users\sdstc\Documents\R\win-library\4.1\00LOCK\data.table\libs\x64\datatable.dll to C:\Users\sdstc\Documents\R\win-library\4.1\data.table\libs\x64\datatable.dll: Permission denied"
Warning message:
"restored 'data.table'"



The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'effects' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'multcomp' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'mvnormtest' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'gapminder' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


Installing package into 'C:/Users/sdstc/Documents/R/win-library/4.1'
(as 'lib' is unspecified)



package 'Ecdat' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\sdstc\AppData\Local\Temp\Rtmp4KmqBV\downloaded_packages


In [7]:
library(readr)
FVH <- read_csv("C:/Users/sdstc/Documents/Education/Other/Entity Academy/Final Project/Final Project/Datasets for Final Project/First Choice Dataset/pivoted_usdaHoney2.csv")
View(FVH)

Rows: 15 Columns: 13
-- Column specification ------------------------------------------------------------------------------------------------
Delimiter: ","
dbl (13): Year, Apples, Fresh, Corn, Fresh Cherries,(Sweet&Sour), Fresh Peac...

i Use `spec()` to retrieve the full column specification for this data.
i Specify the column types or set `show_col_types = FALSE` to quiet this message.


Year Apples, Fresh Corn   Fresh Cherries,(Sweet&Sour)
1  1998 5282.724      247882 336.521                    
2  1999 5746.060      239549 312.226                    
3  2000 5675.890      251854 318.195                    
4  2001 5065.060      241377 339.359                    
5  2002 4605.810      227767 192.320                    
6  2003 4780.600      256278 325.670                    
7  2004 5683.040      299914 353.800                    
8  2005 5261.960      282311 350.170                    
9  2006 5351.610      282303 386.510                    
10 2007 4154.195      338295 407.880                    
11 2008 4508.344      312126 321.315                    
12 2009 4505.000      334052 495.346                    
13 2010 4219.541      339614 367.000                    
14 2011 4231.474      343043 404.660                    
15 2012 4200.000      375683 418.135                    
   Fresh Peaches & Nectarines Grapefruit, Fresh Grapes, Fresh Honey   
1  1248.377                   2280              708.325       99.57162
2  1342.316                   2507              804.873       91.80072
3  1358.464                   2233              822.658       99.58931
4  1297.229                   2199              784.107       84.25344
5  1519.080                   1872              891.164       77.68409
6  1480.070                   1964              730.701       82.26853
7  1429.814                    924              800.663       82.88405
8  1365.400                   1118              903.202       78.91060
9  1171.000                   1476              723.561       69.96081
10 1269.061                   1423              834.909       66.95941
11 1302.563                   1240              893.758       73.92247
12 1194.354                   1123              851.856       65.80139
13 1237.219                   1139              900.934       79.51161
14 1210.412                   1061              896.916       66.76890
15 1099.300                   1092              908.000       63.91401
   Lemons/Limes, Fresh Oranges, Fresh Pears, Fresh Raisins
1  678                  8989          880.061      255.100
2  762                 11875          921.247      315.973
3  904                 11139          884.750      447.877
4  727                 11290          931.631      378.387
5  931                 10527          807.413      402.246
6  724                 11734          847.356      319.238
7  789                  8420          795.837      251.562
8  889                  8394          745.446      324.319
9  724                  7369          754.000      280.774
10 638                  9237          798.200      326.600
11 862                  8487          789.200      354.100
12 800                  7491          867.120      304.400
13 853                  8078          774.477      358.200
14 771                  8180          875.533      348.631
15 827                  8014          796.338      313.795
   Tangerines/Mandarins, Fresh
1  401                        
2  505                        
3  425                        
4  469                        
5  443                        
6  419                        
7  367                        
8  435                        
9  378                        
10 540                        
11 520                        
12 583                        
13 643                        
14 635                        
15 664

In [8]:
#BACKWARD SELECTION
FitAll1 <- lm(Honey ~ ., data = FVH)
FitAll1


Call:
lm(formula = Honey ~ ., data = FVH)

Coefficients:
                  (Intercept)                           Year  
                    1.267e+04                     -6.259e+00  
              `Apples, Fresh`                           Corn  
                    5.291e-03                      1.364e-04  
`Fresh Cherries,(Sweet&Sour)`   `Fresh Peaches & Nectarines`  
                   -5.479e-02                     -3.090e-03  
          `Grapefruit, Fresh`                `Grapes, Fresh`  
                   -2.498e-02                     -1.971e-01  
        `Lemons/Limes, Fresh`               `Oranges, Fresh`  
                    5.201e-02                     -1.691e-04  
               `Pears, Fresh`                        Raisins  
                   -3.576e-02                      9.258e-02  
`Tangerines/Mandarins, Fresh`  
                    1.611e-01  


In [10]:
summary(FitAll1)


Call:
lm(formula = Honey ~ ., data = FVH)

Residuals:
      1       2       3       4       5       6       7       8       9      10 
 2.9093 -1.0572  0.5880 -1.0572 -4.7136  3.1154 -2.6728  4.2037 -2.1135 -2.0352 
     11      12      13      14      15 
 3.8525 -2.6733  0.6032  2.8990 -1.8485 

Coefficients:
                                Estimate Std. Error t value Pr(>|t|)
(Intercept)                    1.267e+04  7.064e+03   1.793    0.215
Year                          -6.259e+00  3.444e+00  -1.818    0.211
`Apples, Fresh`                5.291e-03  1.097e-02   0.482    0.677
Corn                           1.364e-04  2.095e-04   0.651    0.582
`Fresh Cherries,(Sweet&Sour)` -5.479e-02  7.788e-02  -0.704    0.555
`Fresh Peaches & Nectarines`  -3.090e-03  8.276e-02  -0.037    0.974
`Grapefruit, Fresh`           -2.498e-02  3.491e-02  -0.716    0.549
`Grapes, Fresh`               -1.971e-01  1.245e-01  -1.584    0.254
`Lemons/Limes, Fresh`          5.201e-02  6.484e-02   0.802    0.

In [11]:
#The overall p value at the bottom is not significant and  none of the individual
#predictors are, meaning that the model is  not a good fit; what the independent
#variables are does not really matter.

#Once you have created a new model after backward elimination, you can compare the
#above model summary with all 10 predictors against what R has helped determine is the best fit.

In [12]:
step(FitAll1, direction = 'backward')

Start:  AIC=55.96
Honey ~ Year + `Apples, Fresh` + Corn + `Fresh Cherries,(Sweet&Sour)` + 
    `Fresh Peaches & Nectarines` + `Grapefruit, Fresh` + `Grapes, Fresh` + 
    `Lemons/Limes, Fresh` + `Oranges, Fresh` + `Pears, Fresh` + 
    Raisins + `Tangerines/Mandarins, Fresh`

                                Df Sum of Sq    RSS    AIC
- `Oranges, Fresh`               1     0.035 110.60 53.968
- `Fresh Peaches & Nectarines`   1     0.077 110.64 53.974
- `Pears, Fresh`                 1    11.032 121.60 55.390
- `Apples, Fresh`                1    12.864 123.43 55.614
<none>                                       110.57 55.963
- Corn                           1    23.437 134.00 56.847
- `Fresh Cherries,(Sweet&Sour)`  1    27.363 137.93 57.280
- `Grapefruit, Fresh`            1    28.306 138.87 57.383
- `Lemons/Limes, Fresh`          1    35.569 146.13 58.147
- Raisins                        1    59.888 170.46 60.456
- `Tangerines/Mandarins, Fresh`  1   105.024 215.59 63.980
- `Grapes, Fres


Call:
lm(formula = Honey ~ Year + Corn + `Fresh Cherries,(Sweet&Sour)` + 
    `Grapefruit, Fresh` + `Grapes, Fresh` + `Lemons/Limes, Fresh` + 
    Raisins + `Tangerines/Mandarins, Fresh`, data = FVH)

Coefficients:
                  (Intercept)                           Year  
                    1.210e+04                     -5.992e+00  
                         Corn  `Fresh Cherries,(Sweet&Sour)`  
                    1.932e-04                     -4.407e-02  
          `Grapefruit, Fresh`                `Grapes, Fresh`  
                   -1.953e-02                     -1.760e-01  
        `Lemons/Limes, Fresh`                        Raisins  
                    7.398e-02                      5.853e-02  
`Tangerines/Mandarins, Fresh`  
                    1.103e-01  


In [13]:
#According to the the backwards stepwise regression, if none of the predictors
#were removed, it would yield the lowest AIC score of 51.84.

In [14]:
#Create another model for only selected variables and compare their results
fitsome <- lm(formula = Honey ~ Year + Corn + `Fresh Cherries,(Sweet&Sour)` + 
                `Grapefruit, Fresh` + `Grapes, Fresh` + `Lemons/Limes, Fresh` + 
                Raisins + `Tangerines/Mandarins, Fresh`, data = FVH)
fitsome


Call:
lm(formula = Honey ~ Year + Corn + `Fresh Cherries,(Sweet&Sour)` + 
    `Grapefruit, Fresh` + `Grapes, Fresh` + `Lemons/Limes, Fresh` + 
    Raisins + `Tangerines/Mandarins, Fresh`, data = FVH)

Coefficients:
                  (Intercept)                           Year  
                    1.210e+04                     -5.992e+00  
                         Corn  `Fresh Cherries,(Sweet&Sour)`  
                    1.932e-04                     -4.407e-02  
          `Grapefruit, Fresh`                `Grapes, Fresh`  
                   -1.953e-02                     -1.760e-01  
        `Lemons/Limes, Fresh`                        Raisins  
                    7.398e-02                      5.853e-02  
`Tangerines/Mandarins, Fresh`  
                    1.103e-01  


In [15]:
summary(fitsome)


Call:
lm(formula = Honey ~ Year + Corn + `Fresh Cherries,(Sweet&Sour)` + 
    `Grapefruit, Fresh` + `Grapes, Fresh` + `Lemons/Limes, Fresh` + 
    Raisins + `Tangerines/Mandarins, Fresh`, data = FVH)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.2633 -2.3364  0.8131  2.3388  5.3002 

Coefficients:
                                Estimate Std. Error t value Pr(>|t|)   
(Intercept)                    1.210e+04  2.969e+03   4.074  0.00654 **
Year                          -5.992e+00  1.471e+00  -4.073  0.00655 **
Corn                           1.932e-04  1.152e-04   1.676  0.14468   
`Fresh Cherries,(Sweet&Sour)` -4.407e-02  3.927e-02  -1.122  0.30460   
`Grapefruit, Fresh`           -1.953e-02  1.312e-02  -1.489  0.18697   
`Grapes, Fresh`               -1.760e-01  7.295e-02  -2.413  0.05238 . 
`Lemons/Limes, Fresh`          7.398e-02  3.000e-02   2.466  0.04874 * 
Raisins                        5.853e-02  4.794e-02   1.221  0.26787   
`Tangerines/Mandarins, Fresh`  1.103e-01  

In [16]:
#Forward Selection
FitAll2 <- lm(Honey ~ 1, data = FVH)
summary(FitAll2)


Call:
lm(formula = Honey ~ 1, data = FVH)

Residuals:
     Min       1Q   Median       3Q      Max 
-15.0061 -10.4600  -0.0095   4.6487  20.6692 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   78.920      2.984   26.45 2.37e-13 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 11.56 on 14 degrees of freedom


In [17]:
step(FitAll2, direction = 'forward', scope = (formula(FitAll1)))

Start:  AIC=74.39
Honey ~ 1

                                Df Sum of Sq     RSS    AIC
+ Year                           1   1412.18  457.91 55.279
+ Corn                           1   1043.89  826.21 64.132
+ `Grapefruit, Fresh`            1   1043.82  826.28 64.133
+ `Apples, Fresh`                1    991.97  878.12 65.046
+ `Oranges, Fresh`               1    697.15 1172.95 69.388
+ `Tangerines/Mandarins, Fresh`  1    592.55 1277.55 70.670
+ `Fresh Cherries,(Sweet&Sour)`  1    541.50 1328.60 71.257
+ `Grapes, Fresh`                1    424.78 1445.32 72.520
+ `Pears, Fresh`                 1    415.82 1454.28 72.613
+ `Fresh Peaches & Nectarines`   1    377.75 1492.35 73.001
<none>                                       1870.10 74.385
+ Raisins                        1     41.25 1828.85 76.051
+ `Lemons/Limes, Fresh`          1      3.25 1866.85 76.359

Step:  AIC=55.28
Honey ~ Year

                                Df Sum of Sq    RSS    AIC
<none>                                  


Call:
lm(formula = Honey ~ Year, data = FVH)

Coefficients:
(Intercept)         Year  
   4581.701       -2.246  


In [18]:
#StepWise
step(FitAll2, direction = 'both', scope = formula(FitAll1))

Start:  AIC=74.39
Honey ~ 1

                                Df Sum of Sq     RSS    AIC
+ Year                           1   1412.18  457.91 55.279
+ Corn                           1   1043.89  826.21 64.132
+ `Grapefruit, Fresh`            1   1043.82  826.28 64.133
+ `Apples, Fresh`                1    991.97  878.12 65.046
+ `Oranges, Fresh`               1    697.15 1172.95 69.388
+ `Tangerines/Mandarins, Fresh`  1    592.55 1277.55 70.670
+ `Fresh Cherries,(Sweet&Sour)`  1    541.50 1328.60 71.257
+ `Grapes, Fresh`                1    424.78 1445.32 72.520
+ `Pears, Fresh`                 1    415.82 1454.28 72.613
+ `Fresh Peaches & Nectarines`   1    377.75 1492.35 73.001
<none>                                       1870.10 74.385
+ Raisins                        1     41.25 1828.85 76.051
+ `Lemons/Limes, Fresh`          1      3.25 1866.85 76.359

Step:  AIC=55.28
Honey ~ Year

                                Df Sum of Sq     RSS    AIC
<none>                                 


Call:
lm(formula = Honey ~ Year, data = FVH)

Coefficients:
(Intercept)         Year  
   4581.701       -2.246  


In [19]:
fitsome1 <- lm(formula = Honey ~ Year, data = FVH)
summary(fitsome1)


Call:
lm(formula = Honey ~ Year, data = FVH)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.9733 -3.8926 -0.0095  1.7290 11.8204 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 4581.7008   711.1430   6.443 2.19e-05 ***
Year          -2.2458     0.3547  -6.332 2.61e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 5.935 on 13 degrees of freedom
Multiple R-squared:  0.7551,	Adjusted R-squared:  0.7363 
F-statistic: 40.09 on 1 and 13 DF,  p-value: 2.61e-05


In [20]:
#Which model is the best? Why?
bestmodel <- lm(formula = Honey ~ Year + Corn + `Fresh Cherries,(Sweet&Sour)` + 
                  `Grapefruit, Fresh` + `Grapes, Fresh` + `Lemons/Limes, Fresh` + 
                  Raisins + `Tangerines/Mandarins, Fresh`, data = FVH)
bestmodel


Call:
lm(formula = Honey ~ Year + Corn + `Fresh Cherries,(Sweet&Sour)` + 
    `Grapefruit, Fresh` + `Grapes, Fresh` + `Lemons/Limes, Fresh` + 
    Raisins + `Tangerines/Mandarins, Fresh`, data = FVH)

Coefficients:
                  (Intercept)                           Year  
                    1.210e+04                     -5.992e+00  
                         Corn  `Fresh Cherries,(Sweet&Sour)`  
                    1.932e-04                     -4.407e-02  
          `Grapefruit, Fresh`                `Grapes, Fresh`  
                   -1.953e-02                     -1.760e-01  
        `Lemons/Limes, Fresh`                        Raisins  
                    7.398e-02                      5.853e-02  
`Tangerines/Mandarins, Fresh`  
                    1.103e-01  


In [21]:
summary(bestmodel)


Call:
lm(formula = Honey ~ Year + Corn + `Fresh Cherries,(Sweet&Sour)` + 
    `Grapefruit, Fresh` + `Grapes, Fresh` + `Lemons/Limes, Fresh` + 
    Raisins + `Tangerines/Mandarins, Fresh`, data = FVH)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.2633 -2.3364  0.8131  2.3388  5.3002 

Coefficients:
                                Estimate Std. Error t value Pr(>|t|)   
(Intercept)                    1.210e+04  2.969e+03   4.074  0.00654 **
Year                          -5.992e+00  1.471e+00  -4.073  0.00655 **
Corn                           1.932e-04  1.152e-04   1.676  0.14468   
`Fresh Cherries,(Sweet&Sour)` -4.407e-02  3.927e-02  -1.122  0.30460   
`Grapefruit, Fresh`           -1.953e-02  1.312e-02  -1.489  0.18697   
`Grapes, Fresh`               -1.760e-01  7.295e-02  -2.413  0.05238 . 
`Lemons/Limes, Fresh`          7.398e-02  3.000e-02   2.466  0.04874 * 
Raisins                        5.853e-02  4.794e-02   1.221  0.26787   
`Tangerines/Mandarins, Fresh`  1.103e-01  

### From the best model, what is the adjusted R2 value and what does it mean?
### The adjusted R^2 is equal to 0.8214, which ultimately judges whether the data in the regression equation is a good fit. The higher the adjusted R^2, the better the regression equation as it implies that the independent variable is chosen to determine the dependent variable.

### From the best model, how does each variable influence Honey?
### In this case, 82.14% implies that the independent variables explains only that amount of the variation in the dependent variable.


### Which model was the best for each type of method?
### fitsome was the best for this type of method, or lm(formula = Honey ~ Year + Corn + `Fresh Cherries,(Sweet&Sour)` + `Grapefruit, Fresh` + `Grapes, Fresh` + `Lemons/Limes, Fresh` + Raisins + `Tangerines/Mandarins, Fresh`, data = FVH)

### How do the final models from each method compare to each other?
### The AIC of fitsome was 51.84, the smallest out of all of the other models in the stepwise regression, which explains why it was chosen. The summary of the fitsome linear model exhibited characteristics of a "good fit." In other words, the adjusted R^2 value was 82.14%, which means that the independent variable explained nearly 82% or all of the variation of the dependent variable.

### From your chosen "best model," explain what variable(s) contribute to predicting Y and for how much variance they account.
###   Year, Corn, Fresh Cherries, Fresh Grapefruit, Fresh Grapes, Fresh Lemons/Limes, Raisins, and Fresh Tangerines/Mandarins.